sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -924,3 +928,405 @@ def merge_accumulators(self, accumulators):
+  """
+
+  @with_input_types(T)
+  @with_output_types(List[T])
+  class Globally(ptransform.PTransform):
+    """
+    PTransform takes PCollection and returns a list whose single value is
+    approximate N-tiles of the input collection globally.
+
+    Args:
+      num_quantiles: number of elements in the resulting quantiles values list.
+      compare: (optional) Comparator function which is an implementation
+        of "a < b" taking at least two arguments (a and b). Which is later
+        converted to key function as Python 3 does not support cmp.
+      key: (optional) Key is  a mapping of elements to a comparable key, similar
+        to the key argument of Python's sorting methods.
+    """
+
+    def __init__(self, num_quantiles, compare=None, key=None, reverse=False):
+      self.num_quantiles = num_quantiles
+      self.compare = compare
+      self.key = key
+      self.reverse = reverse
+
+    def expand(self, pcoll):
+      return pcoll | core.CombineGlobally(ApproximateQuantilesCombineFn.create(
+          num_quantiles=self.num_quantiles, compare=self.compare,
+          key=self.key, reverse=self.reverse
+      ))
+
+    def display_data(self):
+      return {
+          'num_quantiles': DisplayDataItem(self.num_quantiles,
+                                           label="Quantile Count"),
+          'compare': DisplayDataItem(self.compare.__class__,
+                                     label='Record Comparer FN'),
+          'key': DisplayDataItem(self.key.__class__,
+                                 label='Record Comparer Key'),
+          'reverse': DisplayDataItem(self.reverse.__class__,
+                                     label='Is reversed'),
+      }
+
+  @with_input_types(Tuple[K, V])
+  @with_output_types(List[Tuple[K, V]])
+  class PerKey(ptransform.PTransform):
+    """
+    PTransform takes PCollection of KV and returns a list based on each key
+    whose single value is list of approximate N-tiles of the input element of
+    the key.
+
+    Args:
+      num_quantiles: number of elements in the resulting quantiles values list.
+      compare: (optional) Comparator function which is an implementation
+        of "a < b" taking at least two arguments (a and b). Which is later
+        converted to key function as Python 3 does not support cmp.
+      key: (optional) Key is  a mapping of elements to a comparable key, similar
+        to the key argument of Python's sorting methods.
+    """
+
+    def __init__(self, num_quantiles, compare=None, key=None, reverse=False):
+      self.num_quantiles = num_quantiles
+      self.compare = compare
+      self.key = key
+      self.reverse = reverse
+
+    def expand(self, pcoll):
+      return pcoll | core.CombinePerKey(ApproximateQuantilesCombineFn.create(
+          num_quantiles=self.num_quantiles, compare=self.compare,
+          key=self.key, reverse=self.reverse
+      ))
+
+    def display_data(self):
+      return {
```

Comment:
We return the same format for Globally and PerKey, can we share the format?

Reply:
Done!